### PR TITLE
autotest: test mag fusion

### PR DESCRIPTION
**Pull request by clydemcqueen**
Test several EK3_MAG_CAL values for sub

### CODE REVIEW

**Comment by clydemcqueen** [comment]
@Williangalvani FYI, not hugely important, but may catch future breaking changes

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+/home/clyde/projects/ex_ardu_ign/ap_cmake/ap_master_CMakeLists.txt
```
Comment (peterbarker) [inline CMakeLists.txt:1]: Shouldn't be here?!

File: Tools/autotest/ardusub.py [hunk 2]
```diff
@@ -852,6 +868,58 @@ def backup_home(self):
...
+        self.delay_sim_time(1)
+        self.assert_mag_fusion_selection(MagFuseSel.FUSE_YAW)
+
+        # ALWAYS
+        self.set_parameters({'EK3_MAG_CAL': MagCal.ALWAYS})
+        self.reboot_sitl()
+        self.wait_ready_to_arm()
+        self.assert_mag_fusion_selection(MagFuseSel.FUSE_MAG)
+
+        self.context_pop()
```
Comment (peterbarker) [inline Tools/autotest/ardusub.py:921]:
```suggestion
```

the harness gives you an implicit context

File: Tools/autotest/ardusub.py [hunk 3]
```diff
@@ -852,6 +868,58 @@ def backup_home(self):
...
+            m = mlog.recv_match(type='XKFS')
+            if m is None:
+                break
+            found_sel = m.MAG_FUSION
+        if found_sel != expect_sel:
+            raise NotAchievedException("Expected mag fusion selection %d, found %d" % (expect_sel, found_sel))
+
+    def fuse_mag(self):
+        """Test EK3_MAG_CAL values"""
+        self.context_push()
```
Comment (peterbarker) [inline Tools/autotest/ardusub.py:886]:
```suggestion
```

**Review by peterbarker — CHANGES_REQUESTED** [review]
Nice.

File: CMakeLists.txt (hunk 1 above)
Comment (clydemcqueen) [reply CMakeLists.txt:1]: Doh! Thanks for catching this

File: Tools/autotest/ardusub.py [hunk 4]
```diff
@@ -879,7 +944,8 @@ def tests(self):
             self.MAV_CMD_CONDITION_YAW,
             self.TerrainMission,
             self.SetGlobalOrigin,
-            self.backup_home,
+            self.backup_origin,
+            self.fuse_mag,
```
Comment (peterbarker) [inline Tools/autotest/ardusub.py:948]: Just as a future cleanup, these don't match our naming convention for tests, which you can see in the list above.

Concept is that you can visually distinguish helper functions from tests (kind of entry points, really) that way.

**Review by peterbarker — APPROVED** [review]
LGTM

File: Tools/autotest/ardusub.py (hunk 4 above)
Comment (clydemcqueen) [reply Tools/autotest/ardusub.py:948]: Ack, I was confused about naming, thanks for clearing this up

**Pull request merged**